src/sentry/utils/dates.py

```diff
@@ -102,13 +104,39 @@ def parse_stats_period(period):
     )
 
 
-def get_rollup_from_request(request, params, default_interval, error, top_events=0):
+def get_interval_from_range(date_range: timedelta, high_fidelity: bool) -> str:
+    if date_range >= timedelta(days=60):
+        return "4h" if high_fidelity else "1d"
+
+    if date_range >= timedelta(days=30):
+        return "1h" if high_fidelity else "4h"
+
+    if date_range > timedelta(days=1):
+        return "30m" if high_fidelity else "1h"
+
+    if date_range > timedelta(hours=1):
+        return "1m" if high_fidelity else "5m"
+
+    return "5m" if high_fidelity else "15m"
```

Comment:
Looks good. Matches what we have in the frontend code.

Reply:
I'd like to see if I can completely kill the frontend logic, but not sure if we'll be able to

Reply:
I think it will be hard to remove the frontend as we use the high/low resolution in a few places. We could start removing the usage of the client side code and rely on the server to autogenerate the interval more though.

Reply:
Yeah this was my thought as well